### ISSUE TITLE
Improve `.usa-hint` contrast

### ISSUE THREAD

**Issue by andycochran** [HHS/simpler-grants-gov#10843]
### Summary

In a recent PR (#10807), we discovered that `.usa-character-count__status` does not have enough color contrast because it gets its color from `.usa-hint`. Our `base` palette should be updated to have more accessible defaults. 

> I suspect this ally bug affects more than the character count component. Since `.usa-character-count__status` inherits its color from `.usa-hint`,  I wonder if the `.usa-hint` style is what needs overriding (instead of creating a custom style for `.usa-character-count__status`). All hint text should have the same issue of not being contrasty enough. 
> 
> Unfortunately, there's no theme setting for hint text. The color of `.usa-hint` is [hardcoded](https://github.com/uswds/uswds/blob/e4848f8cf720e7642418a58b0e02c17a51530b53/packages/usa-hint/src/styles/_usa-hint.scss#L5) to `color("base")`. And our theme [overrides](https://github.com/HHS/simpler-grants-gov/blob/main/frontend/src/styles/_uswds-theme.scss#L118-L127) the base colors, setting the middle "base" value to `gray-warm-30` instead of [the default](https://github.com/uswds/uswds/blob/e4848f8cf720e7642418a58b0e02c17a51530b53/packages/uswds-core/src/styles/settings/_settings-color.scss#L31) `gray-cool-50"`: 
> 
> <img width="1007" height="291" alt="image" src="https://github.com/user-attachments/assets/bea5058b-ce23-466e-80d1-94bfd42cf7f2" />
> 
> The default would have enough contrast on white (a magic number ≥50). Perhaps we should consider updating our "base" color palette values. 
> 
> I'd suggest two changes: 
> ```
> $theme-color-base: "gray-warm-50",
> $theme-color-base-dark: "gray-warm-60",
> ```
> 
> I think everything else can stay as-is. But if we do this, we might also do a quick visual audit and make sure nothing else unintentionally breaks. 

### Acceptance criteria

- [ ] Hint text is passes AA contrast 
- [ ] Character count text is passes AA contrast

**Comment by andycochran** [comment]
With these 2 changes… 

```
$theme-color-base: "gray-warm-50",
$theme-color-base-dark: "gray-warm-60",
```

Only visual diffs I notice in an audit are: 

<img width="824" height="207" alt="Image" src="https://github.com/user-attachments/assets/75a43807-e920-4202-b491-2bd1151b0f40" />

^ Slightly darker border on the "Copy search query" box" (not planned, but acceptable) 

----- 

<img width="813" height="295" alt="Image" src="https://github.com/user-attachments/assets/a4b7f01c-72d6-4ef5-a4c3-94fe74c4e6db" />

^ Slightly darker facet counts in filters (not planned, but better a11y) 

----- 

<img width="859" height="248" alt="Image" src="https://github.com/user-attachments/assets/29363894-ef78-4a81-8eac-f7b1877b24a5" />

^ Slightly darker hint text (This is the change we're looking for!) 

----- 

Note: Other text using `.text-base-dark` throughout the app will be 10% darker, such as: 

<img width="971" height="332" alt="Image" src="https://github.com/user-attachments/assets/adf58c2a-68c7-4866-9765-0e39f9c56fe2" />

^ Slightly darker Sendy disclaimer, as it uses `.text-base-dark` (not planned, but better a11y)
